src/app-layer-ssl.c

```diff
@@ -1373,6 +1522,13 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
     int ret;
     uint32_t parsed = 0;
 
+    if (SC_ATOMIC_GET(ssl_config.enable_ja4) &&
+            ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
+        ssl_state->curr_connp->ja4 = SCJA4New();
+    } else {
+        ssl_state->curr_connp->ja4 = NULL;
```

Comment:
Why set it to NULL ?
Either it is NULL already, or the memory is leaking, right ?

Reply:
It's not technically required to set it to NULL and not at all meant as a signal. I was just trying to avoid uninitialized values here in case the `curr_connp` is not zeroed (which it is, I found the associated `memset`). So this can be removed as it only raises questions.